### ISSUE TITLE
[Bug] Sveltekit throws "Syntax Error: Unexpected token 'export'"

### ISSUE THREAD

**Issue by nthskyradiated** [steveruizok/perfect-freehand#75]
I've been getting the the error below in my Sveletekit project:

```
SyntaxError: Unexpected token 'export'
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1153:20)
    at Module._compile (node:internal/modules/cjs/loader:1205:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at cjsLoader (node:internal/modules/esm/translators:283:17)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:233:7)
    at ModuleJob.run (node:internal/modules/esm/module_job:217:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:308:24)
```

The only way I could seem to make it work is by adding another entry to the **package.json** file for svelte:

```javescript
  "exports": {
    ".": {
      "types": "./dist/types/index.d.ts",
      "require": "./dist/cjs/index.js",
      "import": "./dist/esm/index.js",
      "svelte": "./dist/esm/index.js"
    }
,,,

I took some hints from one of Captain Codeman's Svelte-Signature-pad's [commit here.](https://github.com/CaptainCodeman/svelte-signature-pad/commit/577cc4b3af71dd3359634850deeb66e32ab98efd).

**Comment by vudagecko** [comment]
> I've been getting the error below in my Sveltekit project:
> 
> ```
> SyntaxError: Unexpected token 'export'
>     at internalCompileFunction (node:internal/vm:73:18)
>     at wrapSafe (node:internal/modules/cjs/loader:1153:20)
>     at Module._compile (node:internal/modules/cjs/loader:1205:27)
>     at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
>     at Module.load (node:internal/modules/cjs/loader:1091:32)
>     at Module._load (node:internal/modules/cjs/loader:938:12)
>     at cjsLoader (node:internal/modules/esm/translators:283:17)
>     at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:233:7)
>     at ModuleJob.run (node:internal/modules/esm/module_job:217:25)
>     at async ModuleLoader.import (node:internal/modules/esm/loader:308:24)
> ```
> 
> The only way I could seem to make it work is by adding another entry to the **package.json** file for svelte:
> 
> ```
>   "exports": {
>     ".": {
>       "types": "./dist/types/index.d.ts",
>       "require": "./dist/cjs/index.js",
>       "import": "./dist/esm/index.js",
>       "svelte": "./dist/esm/index.js"
>     }
> ```
> 
> I took some hints from one of Captain Codeman's Svelte-Signature-pad's [commit here.](https://github.com/CaptainCodeman/svelte-signature-pad/commit/577cc4b3af71dd3359634850deeb66e32ab98efd).

Are you able to build with your workaround mentioned above? Even with that added, the build fails for me in Sveltekit.

Does anyone have a fix for this? I am getting the same error mentioned above.

**Comment by nthskyradiated** [comment]
yes but while it does work, the problem I faced was installing the fix upon deployment on vercel. I ended up repackaging it on npm just so I could use it on my project. unless the fix is applied by the author, there's just no way around I could think of.

Feel free to use it. [here](https://www.npmjs.com/package/perfect-freehand-for-svelte)

**Comment by ali-garajian** [comment]
> yes but while it does work, the problem I faced was installing the fix upon deployment on vercel. I ended up repackaging it on npm just so I could use it on my project. unless the fix is applied by the author, there's just no way around I could think of.
> 
> Feel free to use it. [here](https://www.npmjs.com/package/perfect-freehand-for-svelte)

I think you could've also used something like [patch-package](https://www.npmjs.com/package/patch-package)

**Comment by steveruizok** [comment]
The package exports have been updated in recent versions and should now work correctly with SvelteKit. The current `package.json` has proper ESM/CJS dual exports:

```json
"exports": {
  ".": {
    "types": "./dist/types/index.d.ts",
    "require": "./dist/cjs/index.js",
    "import": "./dist/esm/index.mjs"
  }
}
```

The ESM output now uses the `.mjs` extension which helps module resolution. If you're still having issues with SvelteKit, please ensure you're using version 1.2.3 or later and feel free to reopen this issue with reproduction steps.
